src/services/completions.ts

```diff
@@ -2121,10 +2123,21 @@ export function getCompletionEntriesFromSymbols(
             }
             // Filter out variables from their own initializers
             // `const a = /* no 'a' here */`
-            if (variableDeclaration && symbol.valueDeclaration === variableDeclaration) {
+            if (tryCast(variableOrParameterDeclaration, isVariableDeclaration) && symbol.valueDeclaration === variableOrParameterDeclaration) {
                 return false;
             }
 
+            // Filter out parameters from their own initializers
+            // `function f(a = /* no 'a' and 'b' here */, b) { }`
+            const symbolDeclaration = symbol.valueDeclaration;
+            if (tryCast(variableOrParameterDeclaration, isParameter) && tryCast(symbolDeclaration, isParameter)) {

```

Comment:
I would just write `variableOrParameterDeclaration && isParameter(variableOrParameterDeclaration) && symbolDeclaration && isParameter(symbolDeclaration)` just so you can avoid the type assertions below.

Reply:
Not a problem!